src/m365/spo/commands/tenant/tenant-applicationcustomizer-get.spec.ts

```diff
@@ -0,0 +1,213 @@
+    "FileSystemObjectType": 0,
+    "Id": 4,
+    "ServerRedirectedEmbedUri": null,
+    "ServerRedirectedEmbedUrl": "",
+    "ContentTypeId": "0x00693E2C487575B448BD420C12CEAE7EFE",
+    "Title": title,
+    "Modified": "2023-01-11T15:47:38Z",
+    "Created": "2023-01-11T15:47:38Z",
+    "AuthorId": 9,
+    "EditorId": 9,
+    "OData__UIVersionString": "1.0",
+    "Attachments": false,
+    "GUID": id,
+    "ComplianceAssetId": null,
+    "TenantWideExtensionComponentId": clientSideComponentId,
+    "TenantWideExtensionComponentProperties": "{\"testMessage\":\"Test message\"}",
+    "TenantWideExtensionWebTemplate": null,
+    "TenantWideExtensionListTemplate": 0,
+    "TenantWideExtensionLocation": "ClientSideExtension.ApplicationCustomizer",
+    "TenantWideExtensionSequence": 0,
+    "TenantWideExtensionHostProperties": null,
+    "TenantWideExtensionDisabled": false
+  }];
+
+  let log: any[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      Cli.executeCommandWithOutput
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.TENANT_APPLICATIONCUSTOMIZER_GET);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if the id is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: 'abc' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the clientSideComponentId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { clientSideComponentId: 'abc' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if id is a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: id } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('handles error when app catalog not registered', async () => {
+    const errorMessage = 'No app catalog URL found';
+    sinon.stub(Cli, 'executeCommandWithOutput').callsFake(async (command): Promise<any> => {
```

Comment:
Rather than stubbing commands, have you tried seeing if it's possible to stub the request that's issued towards the API? That way, we'd be less dependent on the current command implementation (executing other commands vs. calling the API directly).

Reply:
Hi @waldekmastykarz 
I tried stubbing the request, but did not work.

Below is the test case I have tried.

```
it('retrieves an application customizer by title', async () => {
    sinon.stub(request, 'get').callsFake(async (opts) => {
      requests.push(opts);
      if ((opts.url as string).indexOf('SP_TenantSettings_Current') > -1) {
        return { stdout: JSON.stringify(appCatalogUrl) };
      }
      throw 'Invalid request';
    });
.
.
.
```

Reply:
What's failing exactly?

Reply:
Hi @waldekmastykarz 
Request does not fail. But, it does not provide the code coverage for ts file.

Reply:
Could you share more information about this? What's not being covered?